pkgs/development/python-modules/devito/update.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i python -p python39Packages.requests
+
+import requests
+import subprocess
+import json
+
+def grabLatestCommitMetaData(pkg):
+     nixPrefetch = json.loads(subprocess.check_output(["nix-prefetch-git", f"https://github.com/{pkg}"]).decode("utf-8").strip('\n'))
+     return nixPrefetch
+
+def findLine(key,derivation):
+    count = 0
+    lines = []
+    for line in derivation:
+        if key in line:
+            lines.append(count)
+        count += 1
+    return lines
+
+def updateVersion(pkg,derivation):
+    key = "version = "
+    line = findLine(key,derivation)[0]
+    derivation[line] = f"""  version = "unstable-{pkg['date'][:10]}"\n;"""
+
+    key = "rev = "
+    line = findLine(key,derivation)[0]
+    derivation[line] = f"""    rev = "{pkg['rev']}"\n;"""
+    return derivation
+
+def updateHashes(pkg,derivation):
+    key = "sha256 = "
+    line = findLine(key,derivation)[0]
+    derivation[line] = f"""    sha256 = "{pkg['sha256']}"\n;"""
+    return derivation
+
+def grabRequirements_txt(url):
+    return requests.get(url).text.split('\n')
+
+def updateRequirementsPatch(requirements,derivation):
+    key = "distributed"
+    lineReq = findLine(key,requirements)[0]
+    lineDer = findLine(key,derivation)[0]
+    derivation[lineDer] = f"""    --replace "{requirements[lineReq]}" "distributed" \ \n;"""
+
+    key = "flake8"
+    lineReq = findLine(key,requirements)[0]
+    lineDer = findLine(key,derivation)[0]
+    derivation[lineDer] = f"""    --replace "{requirements[lineReq]}" "" \ \n;"""
+
+    key = "pytest"
+    lineReq = findLine(key,requirements)[0]
+    lineDer = findLine(key,derivation)[0]
+    derivation[lineDer] = f"""    --replace "{requirements[lineReq]}" "" \ \n;"""
+
+    key = "pytest-runner"
+    lineReq = findLine(key,requirements)[0]
+    lineDer = findLine(key,derivation)[0]
+    derivation[lineDer] = f"""    --replace "{requirements[lineReq]}" "" \ \n;"""
+
+    key = "pytest-cov"
+    lineReq = findLine(key,requirements)[0]
+    lineDer = findLine(key,derivation)[0]
+    derivation[lineDer] = f"""    --replace "{requirements[lineReq]}" "" \ \n;"""
+
+    return derivation
```

Comment:
This could be easily replaced with a one sed script that removes everything after the first match of ``[<>]=``.

Reply:
Do you have any example of how could this  be done?

Reply:
It would be very nice to have an example haha.

Reply:
@SuperSandro2000 I discussed this with some colleagues, and I find that the sed approach would be too easy to break, while my update script, although I admit is a bit overengineered, it is far more robust than a sed.

Reply:
https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/mitmproxy/default.nix#L100 

https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/kaptan/default.nix#L18

Reply:
Yes, the script is very over engineered and as soon as it breaks the next person will most likely just remove it.

Reply:
@SuperSandro2000 I adhered to your suggestions, pls do check if is ok now.